cicecore/cicedyn/infrastructure/ice_restoring.F90

```diff
@@ -88,20 +104,28 @@ subroutine ice_HaloRestore_init
 
    if ((ew_boundary_type == 'open' .or. &
         ns_boundary_type == 'open') .and. .not.(restart_ext)) then
-      if (my_task == master_task) write (nu_diag,*) ' ERROR: restart_ext=F and open boundaries'
+      if (my_task == master_task) write (nu_diag,*) 'ERROR: restart_ext=F and open boundaries'
       call abort_ice(error_message=subname//'open boundary and restart_ext=F', &
          file=__FILE__, line=__LINE__)
    endif
 
    allocate (aicen_rest(nx_block,ny_block,ncat,max_blocks), &
              vicen_rest(nx_block,ny_block,ncat,max_blocks), &
              vsnon_rest(nx_block,ny_block,ncat,max_blocks), &
-             trcrn_rest(nx_block,ny_block,ntrcr,ncat,max_blocks))
-
-   aicen_rest(:,:,:,:) = c0
-   vicen_rest(:,:,:,:) = c0
-   vsnon_rest(:,:,:,:) = c0
-   trcrn_rest(:,:,:,:,:) = c0
+             trcrn_rest(nx_block,ny_block,ntrcr,ncat,max_blocks),&
+             uvel_rest(nx_block,ny_block,max_blocks),&  !pedrocice
+             vvel_rest(nx_block,ny_block,max_blocks))
+
+!    aicen_rest(:,:,:,:) = c0
```

Comment:
@SmithJos13 it seems you only add two more variables. Please clean this part to have diff only for your modifications.